web3/contract/async_contract.py

```diff
@@ -577,7 +580,10 @@ def __init__(
             if transaction is None:
                 transaction = {}
 
-            self._functions = filter_by_type("function", self.abi)
+            self._functions = [
+                cast(ABIFunction, fn_abi)
```

Comment:
I don't really have any ideas other than making them separate functions - for example: calling this something like `def get_all_functions(self.abi) -> ABIFunction:`- but I wonder if there is a way that we could generally stop casting to `ABIFunction`/`ABIEvent` when we call `filter_abi_by_type`. It feels a little smelly, but I'm not hugely concerned about it either.

Reply:
🤦 Oh you're right! I can simply change this to the new `get_all_function_abis` and update the type of `_functions`.